### ISSUE TITLE
players can still have rifle when its pistol round 

### ISSUE THREAD

**Issue by GTAmama16** [LordFetznschaedl/CS2Retake#4]
hey there, first of all thanks for the great plugin, but Im having issues when its pistol round but players are spawn with rifles

**Comment by LordFetznschaedl** [comment]
Hello @GTAmama16 
Thank you for your feedback.

This is an issue with the weaponKits.json config.

All available weapon kits currently have the RoundType "Undefined" which is a fallback if no weapon kit is configured for the specific roundtype. This was done to save on time while developing and most likely a better default config will be added in the future.

If you want to have weapon kits that are roundtype dependent you have the ability to edit this config.

Following RoundTypes are currently available:
- Undefined (used as a fallback)
- Pistol
- Mid
- FullBuy

I hope this helped.

Thank you and regards,
Robert

**Comment by GTAmama16** [comment]
@LordFetznschaedl  Thanks for the reply! I will check the weapoinkits.json file later, and Im still having one more question 
when Im adding the spawn into the maps , will it be possible for me to set some certain spawns for Terrorist as the "bomb holder" only spawn ?

**Comment by LordFetznschaedl** [comment]
This is a valid question. 
No this is not possible. I decided against a system where you can define if a spawn is bomb carrier only. All spawns within the bombzone are valid spawns for a spawn carrier.

Maybe if enough people wish for a system like this I might extend it.

**Comment by GTAmama16** [comment]
@LordFetznschaedl  Cool thanks for the reply, but when Im adding the spawn manually, whether im in the bomb site or not, it always show "IsInBombZone:false"

**Comment by LordFetznschaedl** [comment]
@GTAmama16 
Ideally you could send me the map and a couple of spawn coordinates so I can try and find if an error exists.
It could be that one of the last releases of CSS broke something

**Comment by GTAmama16** [comment]
@LordFetznschaedl Sure so on the mirage bombsite A terrorist spawns are 

> 
{"SpawnId":"945be0f8-9f46-4e9f-a820-dda0e91b96ee","Team":2,"BombSite":0,"IsInBombZone":false,"PositionX":-525.559,"PositionY":-2194.9321,"PositionZ":-179.96875,"QAngleX":0,"QAngleY":24.739151,"QAngleZ":0}

> {"SpawnId":"e8a6200f-9dd7-4114-a8bb-b4000843c174","Team":2,"BombSite":0,"IsInBombZone":false,"PositionX":-264.7378,"PositionY":-2175.1824,"PositionZ":-171.54071,"QAngleX":0,"QAngleY":53.412445,"QAngleZ":0}


It will show out the message 
>SpawnPoint added! BombSite: A - Team: Terrorist - isInBombZone: False

Bombsite B 

> {"SpawnId":"32278846-5907-49e3-97e2-2093def675aa","Team":2,"BombSite":0,"IsInBombZone":false,"PositionX":-2033.1505,"PositionY":261.1696,"PositionZ":-159.96875,"QAngleX":0,"QAngleY":82.93921,"QAngleZ":0}

> {"SpawnId":"71d6d0f5-1a16-406a-9a03-6e27db7c192e","Team":2,"BombSite":0,"IsInBombZone":false,"PositionX":-2011.0289,"PositionY":388.80957,"PositionZ":-159.96875,"QAngleX":0,"QAngleY":57.216797,"QAngleZ":0}

thx for the reply !

**Comment by t1ckrate** [comment]
@GTAmama16 I confirm that i've experienced this bug. I've managed to fix it by putting the spawns during a round in T side (not in warmup) while holding the bomb. Not sure if this will work for you but it seems to be the best situation to made it work.

**Comment by GTAmama16** [comment]
@t1ckrate thx for the reply ! , I did tried to swap to T side when adding the spawns , but whenever I switch over to T side the round will just reset and put me back to CT again , I add all the spawn when I’m alone in server without any other players

**Comment by t1ckrate** [comment]
@GTAmama16 If you weren't able to place any spawn that has isInBombZone to true, it might be the problem that none of your friend will get the bomb. Maybe has a workaround for now you could edit manually the `spawns/de_mirage.json` file and edit the A spawn in bomb area from false to true.

**Comment by t1ckrate** [comment]
@GTAmama16 Yes, in the first place you have to use the `retake.cfg` file which is located here [https://github.com/LordFetznschaedl/CS2-Configs/tree/master/retake](url) to maximize your chances that the plugin works.

**Comment by GTAmama16** [comment]
@t1ckrate thx for the reply , I did use these configs for this plugin , seems like I should delete the map spawn file and manually adding it again , or is it possible for you to send me the spawn config file so I can test it on my server ?

**Comment by t1ckrate** [comment]
@GTAmama16 You're welcome. You will find the following file that i use for my retake server, spawns aren't perfect but this works.

`[{"SpawnId":"021b078e-50c5-48e7-b1af-80069c13783d","Team":2,"BombSite":0,"IsInBombZone":false,"PositionX":-87.07378,"PositionY":-1665.3823,"PositionZ":-167.96875,"QAngleX":0,"QAngleY":-158.72498,"QAngleZ":0},{"SpawnId":"08453520-fa84-4fdc-a5c1-375618f1053e","Team":2,"BombSite":0,"IsInBombZone":false,"PositionX":-199.52673,"PositionY":-1659.2239,"PositionZ":-167.96875,"QAngleX":0,"QAngleY":175.03555,"QAngleZ":0},{"SpawnId":"6d1a509b-05ec-4cea-a863-4f33db3d66aa","Team":2,"BombSite":0,"IsInBombZone":false,"PositionX":-150.30559,"PositionY":-2256.577,"PositionZ":-167.96875,"QAngleX":0,"QAngleY":135.84491,"QAngleZ":0},{"SpawnId":"11c41ab5-46a9-4721-b1e6-8b22e04004be","Team":2,"BombSite":0,"IsInBombZone":true,"PositionX":-563.0255,"PositionY":-2132.649,"PositionZ":-179.96875,"QAngleX":0,"QAngleY":165.24432,"QAngleZ":0},{"SpawnId":"252ea020-854c-4db0-83cb-c432dc790e4a","Team":2,"BombSite":0,"IsInBombZone":true,"PositionX":-628.3371,"PositionY":-1992.3688,"PositionZ":-179.96875,"QAngleX":0,"QAngleY":-4.6338196,"QAngleZ":0},{"SpawnId":"5851bdf9-048d-47d9-be9b-d51c68edbb19","Team":2,"BombSite":0,"IsInBombZone":false,"PositionX":-496.03125,"PositionY":-1309.0312,"PositionZ":-159.96875,"QAngleX":0,"QAngleY":-164.3692,"QAngleZ":0},{"SpawnId":"c2d94544-ee38-447a-9f05-a420a8790817","Team":2,"BombSite":0,"IsInBombZone":false,"PositionX":-992.03033,"PositionY":-1304.8324,"PositionZ":-155.61627,"QAngleX":0,"QAngleY":-154.8626,"QAngleZ":0},{"SpawnId":"9ca75050-c3b5-4376-9fcd-88e448731494","Team":3,"BombSite":0,"IsInBombZone":false,"PositionX":-1881.3091,"PositionY":-611.546,"PositionZ":-167.96875,"QAngleX":0,"QAngleY":-14.505386,"QAngleZ":0},{"SpawnId":"6e5addae-4120-4a9d-92f9-07b2d9215399","Team":3,"BombSite":0,"IsInBombZone":false,"PositionX":-998.0216,"PositionY":279.67398,"PositionZ":-367.96875,"QAngleX":0,"QAngleY":-89.21036,"QAngleZ":0},{"SpawnId":"bc5635ba-9602-4f13-8dc5-fc79e27fcf1a","Team":3,"BombSite":0,"IsInBombZone":false,"PositionX":292.7391,"PositionY":-825.7071,"PositionZ":-163.7995,"QAngleX":0,"QAngleY":173.31378,"QAngleZ":0},{"SpawnId":"16f123fd-d40a-418a-b9b3-5a2944202571","Team":3,"BombSite":0,"IsInBombZone":false,"PositionX":-996.6149,"PositionY":162.20395,"PositionZ":-172.21704,"QAngleX":0,"QAngleY":-7.3560333,"QAngleZ":0},{"SpawnId":"39549f93-b027-47f5-a6ec-d2bf79b8444f","Team":2,"BombSite":1,"IsInBombZone":true,"PositionX":-1987.0312,"PositionY":385.08423,"PositionZ":-159.96875,"QAngleX":0,"QAngleY":108.83847,"QAngleZ":0},{"SpawnId":"54c14dc1-8491-46a5-8e08-d7f87e9e7371","Team":2,"BombSite":1,"IsInBombZone":true,"PositionX":-1943.0662,"PositionY":242.0312,"PositionZ":-159.96875,"QAngleX":0,"QAngleY":40.27484,"QAngleZ":0},{"SpawnId":"9af6ef80-196b-4c6e-a047-880cdbd6c395","Team":2,"BombSite":1,"IsInBombZone":true,"PositionX":-2198.9688,"PositionY":89.03124,"PositionZ":-159.96875,"QAngleX":0,"QAngleY":75.66318,"QAngleZ":0},{"SpawnId":"fafae836-a3de-4fb4-a6c6-dcb20af50a61","Team":2,"BombSite":1,"IsInBombZone":false,"PositionX":-2494.5312,"PositionY":284.69592,"PositionZ":-167.96869,"QAngleX":0,"QAngleY":-2.494934,"QAngleZ":0},{"SpawnId":"b1c86a92-cd4f-44f0-a155-71c0d8359756","Team":2,"BombSite":1,"IsInBombZone":false,"PositionX":-2415.9714,"PositionY":-231.90942,"PositionZ":-164.11218,"QAngleX":0,"QAngleY":-11.956909,"QAngleZ":0},{"SpawnId":"b61cd677-f8b2-4aa3-8420-e1a54139ba45","Team":2,"BombSite":1,"IsInBombZone":false,"PositionX":-1614.6841,"PositionY":-179.7771,"PositionZ":-166.27917,"QAngleX":0,"QAngleY":173.97916,"QAngleZ":0},{"SpawnId":"73d436ae-50a9-496c-aa17-b8abef667c70","Team":2,"BombSite":1,"IsInBombZone":false,"PositionX":-2058.861,"PositionY":838.5316,"PositionZ":-47.252502,"QAngleX":0,"QAngleY":-45.518066,"QAngleZ":0},{"SpawnId":"3a92754e-e99f-4f9f-94ea-839c491362e0","Team":3,"BombSite":1,"IsInBombZone":false,"PositionX":55.75819,"PositionY":788.614,"PositionZ":-135.96875,"QAngleX":0,"QAngleY":-177.83124,"QAngleZ":0},{"SpawnId":"21db6639-3643-4787-b732-86f79a8cc5c6","Team":3,"BombSite":1,"IsInBombZone":false,"PositionX":-979.16943,"PositionY":440.51416,"PositionZ":-367.96875,"QAngleX":0,"QAngleY":12.419708,"QAngleZ":0},{"SpawnId":"44f8571f-63a1-4d16-8115-86e288475a78","Team":3,"BombSite":1,"IsInBombZone":false,"PositionX":-417.50153,"PositionY":-404.56842,"PositionZ":-167.10449,"QAngleX":0,"QAngleY":173.05527,"QAngleZ":0},{"SpawnId":"3b3b6ed6-97a1-40ea-914a-eb279f505e55","Team":3,"BombSite":1,"IsInBombZone":false,"PositionX":-1823.1438,"PositionY":-1902.9493,"PositionZ":-264.80078,"QAngleX":0,"QAngleY":74.71219,"QAngleZ":0}]`

**Comment by GTAmama16** [comment]
@t1ckrate thx for the help , rly appreciate it , is the spawn for mirage ? 
I will test it tomorrow , and maybe I will find a way that I can spawn as T with bomb on my hand , btw do you run the server on competitive mode ?
Or do I have to change it to other game mode 
Thanks for the help !

**Comment by t1ckrate** [comment]
@GTAmama16 Yes the spawn file is for `de_mirage` !
My server is currently running on competitive mode, and you've to place the `retake.cfg` file (provided in the repo earlier) in `csgo/cfg/cs2retake/` folder (you'll need to create it first).

**Comment by GTAmama16** [comment]
@t1ckrate thanks for the respawn config ! It did work on my server, but im facing some issues , just wondering r u having these issues too ? 
I cant assign team when the server is empty , I will have to wait for the 15 secs then it will automatically assign me to CT, and  when im playing as terrorist I wont hold bomb until I move my character , sometime its a bit confused for players that cant tell if their holding the bomb or not

**Comment by LordFetznschaedl** [comment]
I have noticed this as well yesterday while testing a bit (didnt have a lot of time).
It seems that the queuing system is now working since version 61 of CSS.
I will have to look into it more closely.

**Comment by GTAmama16** [comment]
@LordFetznschaedl  Yeah im running latest CSS plugins atm, still having issues when auto queueing and auto holding bomb when spawn , btw I have a few suggestions 
1: having cooldown time for T to plant the bomb before CT can move 
2: making max players as 9 4T and 5CT for better gameplay experience, or make it customizable 
3: having T auto hold bomb  when hes a bomb holder 
4: Having center to notify player when its A 

just find another issue when im testing with 10 players, team wont auto scramble when we first join the server, so we will have to manually choose our side to make it 5v5

**Comment by LordFetznschaedl** [comment]
Thank you for the suggestions:
1) I hope that the planting yourself feature can soon be replaced by a working autoplant. This is one of the priorities currently.
2) Advanced settings are planned for the future. Regarding the max players: You should be able to set it using the -maxplayers start parameter of your server.
The plugins logic should still work if you set it to 7 players for example.
3) I did try todo this but at least it the version of CSS when I implemented it, it wasnt possible (or I didnt get it to work)
4) The important gameinfo like Spot, Rountype and if you have to plant will come as a center message

I hope I could give you a bit of insides 🙂

Sadly this week I have not that much time to make a lot of progress.

Regards, 
Robert

**Comment by GTAmama16** [comment]
@LordFetznschaedl Thx for the reply, first of all take no rush for implementing those settings , its great to see that retake plugins are back in CS2, its great to hear that those suggestions will be add in the future

I have one more suggestion which I think its the most important one for now , for the `retakeaddspawn` command , will it be possible for it to show a setting menu which we can check for nearest spawn location, creating new T/CT spawn , or having ability to delete spawns, cuz theres multiple time that I accidently adding the wrong spawn on my map, and the only way that I can delete it, its by deleting it in the spawn config. Which sometime its a bit confuse since I cant check the X,Y,Z position on the client side 

Regard

**Comment by LordFetznschaedl** [comment]
Hello @GTAmama16 ,

I am sorry that I didnt see your suggestion earlier. 
Yes an Editor mode is planned for the future.
